### PR TITLE
RHDM-1430 Add new reference implementation for Optaplanner School Time Table

### PR DESCRIPTION
**Pull request by Ginxo**
https://issues.redhat.com/browse/RHDM-1430

**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[link](https://www.example.com)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

paste the link(s) from GitHub here

**How to retest or run**:

* a pull request please add comment: regex **[.\*[j|J]enkins,?.\*(retest|test) this.\*]**

* a full downstream build please add comment: regex **[.*\[j|J]enkins,?.\*(execute|run|trigger|start|do) fdb.\*]**

* a compile downstream build please  add comment: regex **[.\*[j|J]enkins,?.\*(execute|run|trigger|start|do) cdb.\*]**

* a full production downstream please add comment: regex **[.\*[j|J]enkins,?.\*(execute|run|trigger|start|do) product fdb.\*]**

* an upstream build please add comment: regex **[.\*[j|J]enkins,?.\*(execute|run|trigger|start|do) upstream.\*]**

i.e for running a full downstream build =  **Jenkins do fdb**

### CODE REVIEW

**Comment by Ginxo** [comment]
Jenkins execute product fdb

File: optaplanner-quickstarts/pom.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
   <url>https://www.optaplanner.org</url>
 
   <modules>
+    <module>quarkus-school-timetabling</module>
```
Comment (mareknovotny) [inline optaplanner-quickstarts/pom.xml:26]: Quarkus should not be part of default build on branch 7.x
Better to remove it completely and keep it only on master branch, wdyt @ge0ffrey @triceo @rsynek ?

File: optaplanner-quickstarts/quarkus-school-timetabling/pom.xml [hunk 2]
```diff
@@ -13,6 +13,11 @@
     <version.org.optaplanner>7.43.0-SNAPSHOT</version.org.optaplanner>
     <version.io.quarkus>1.7.0.Final</version.io.quarkus>
 
+    <version.org.webjars.bootstrap>4.3.1</version.org.webjars.bootstrap>
+    <version.org.webjars.jquery>3.4.1</version.org.webjars.jquery>
+    <version.org.webjars.font-awesome>5.11.2</version.org.webjars.font-awesome>
+    <version.org.webjars.momentjs>2.24.0</version.org.webjars.momentjs>
+
```
Comment (mareknovotny) [inline optaplanner-quickstarts/quarkus-school-timetabling/pom.xml:20]: fyi there were already merged this PR https://github.com/kiegroup/optaplanner/pull/898 which handles the movement of hardcoded dependencyversions to version property, so just rebase this PR

**Review by mareknovotny — CHANGES_REQUESTED** [review]
just rebase from 7.x branch and also I think quarkus quickstart should be removed completely on 7.x branch, but wait with that on @ge0ffrey or anybody from Optaplanner team

File: optaplanner-quickstarts/quarkus-school-timetabling/pom.xml (hunk 2 above)
Comment (Ginxo) [reply optaplanner-quickstarts/quarkus-school-timetabling/pom.xml:20]: @mareknovotny it was done only for `optaplanner-quickstarts/spring-boot-school-timetabling` so I decided to apply same pattern for quarkus one

**Comment by Ginxo** [comment]
Jenkins execute product fdb

**Review by triceo — COMMENTED** [review]
If it's all the same to you, we'll keep the Quarkus quickstart. We want community users to be able to start using it, and `8.0.0.Final` is still a long time away.

That said, I will leave the final decision up to @ge0ffrey .

**Comment by mareknovotny** [comment]
@triceo if that quarkus quickstart is not part of default build that is enough, but I still believe you are mixing apples with oranges as quarkus quickstart on v7 branch is not targeting correct branch.

File: pom.xml [hunk 3]
```diff
@@ -34,6 +34,7 @@
     <!-- TODO https://issues.redhat.com/browse/PLANNER-2042 -->
     <spotbugs.failOnViolation>false</spotbugs.failOnViolation>
     <version.org.mockito>3.1.0</version.org.mockito>
+    <optaplanner-spring-boot-school-timetabling-quickstart.version>0.1.0-SNAPSHOT</optaplanner-spring-boot-school-timetabling-quickstart.version>
```
Comment (mareknovotny) [inline pom.xml:37]: one additional comment to standards of property names. I would suggest to use always the same pattern
version.groupId[.more-identifying-id]

Comment (Ginxo) [reply pom.xml]: done, rnamed to `version.com.example.optaplanner-school-timetabling-quickstart`

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**